lib/orbit/import/rfid.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Orbit.Import.Rfid do
+  require Logger
+  alias Orbit.Employee
+  alias Orbit.Repo
+  alias Orbit.S3
+
+  @spec download() :: binary()
+  def download do
+    {:ok, csv_data} =
+      S3.read(
+        :glides_global,
+        Keyword.fetch!(Application.fetch_env!(:orbit, Orbit.Import.Rfid), :pathname)
+      )
+
+    csv_data
+  end
+
+  @badge_number_field "EMPLOYEE_ID"
+  @badge_serial_field "SERIAL_NUMBER"
+
+  @spec import_rows(Enumerable.t()) :: :ok
+  def import_rows(rows) do
+    imported_count =
+      rows
+      |> Enum.map(fn row ->
+        Map.update!(row, @badge_number_field, &String.trim_leading(&1, "0"))
+      end)
+      |> Enum.group_by(& &1[@badge_number_field])
+      |> Enum.reduce(
+        0,
+        fn {badge_number, badge_serials}, acc ->
+          count =
+            case Repo.get_by(Employee, badge_number: badge_number) do
+              nil ->
+                0
+
+              employee ->
+                {:ok, employee} =
+                  employee
+                  |> Repo.preload([:badge_serials])
+                  |> Employee.changeset(%{
+                    badge_serials:
+                      Enum.map(
+                        badge_serials,
+                        &%{badge_serial: &1[@badge_serial_field]}
```

Comment:
This will still work if the serial number is an empty string, right?

Reply:
I think it would just insert a row with empty string, but from looking at the file there aren't any blanks in either of the two columns we're using here.